hashicorp/dev-portal
# Web Util Deprecation: Remove `<NewsletterSignupForm />`

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-nels-webutil-server-deprecation-8a6c4b-hashicorp.vercel.app/sentinel) 🔎
- [Asana task](https://app.asana.com/0/1202802428287999/1203229194220543/f) 🎟️

## 🗒️ What



This PR removes `<NewsletterSignupForm />` thereby removing `WebUtilServer` dependencies from Dev Portal. 
- Removes `<NewsletterSignupForm />`
- Removes `<NewsletterSignupForm />` from footer
- Removes packages that are specific to `<NewsletterSignupForm />` ([react-text-input](https://hashicorp-mktg.sourcegraph.com/search?q=context%3Aglobal+repo%3A%5Egithub%5C.com%2Fhashicorp%2Fdev-portal%24+react-text-input&patternType=standard&groupBy=path), [formik](https://hashicorp-mktg.sourcegraph.com/search?q=context%3Aglobal+repo%3A%5Egithub%5C.com%2Fhashicorp%2Fdev-portal%24+formik&patternType=standard&groupBy=path), [react-checkbox-input](https://hashicorp-mktg.sourcegraph.com/search?q=context%3Aglobal+repo%3A%5Egithub%5C.com%2Fhashicorp%2Fdev-portal%24+react-checkbox-input&patternType=standard&groupBy=path)).

## 🤷 Why



In 2023, we will be deprecating the `WebUtilServer`. Across our various codebases, there are a number of forms that utilize the `WebUtilServer`'s `forms/standard` endpoint.

## 🛠️ How



As mentioned above, this PR removes the `<NewsletterSignupForm />`. In dot com, we refactored `<NewsletterSignupForm />` to use the `<MarketoForm />` component. Initially, the assumption was that we could follow suit here: simply replace the current implementation of  `<NewsletterSignupForm />` with the newly refactored version in dot com. However, it was overlooked that this would require one to copy all of the Marketo API endpoint logic here in Dev Portal.

Ultimately, it was decided that we would simply remove the component entirely instead of copying a fair amount of logic. The reasons for this are as follows: a fair amount of logic would need to be copied into Dev Portal. More copied code means more code to maintain and a high risk of drift. Additionally, this form only renders in the footer of the /sentinel page. Given its extremely limited usage, it was decided that the juice wasn't worth the squeeze.

## 📸 Design Screenshots



## 🧪 Testing



- Navigate to the [Preview](dev-portal-git-nels-webutil-server-deprecation-8a6c4b-hashicorp.vercel.app/)
- Select "Sentinel" using the product selector (?)
- Navigate to the Sentinel docs [page](https://dev-portal-git-nels-webutil-server-deprecation-8a6c4b-hashicorp.vercel.app/sentinel)
- Scroll to the footer
- Verify no visual regressions
- Verify that the form has been removed

## 💭 Anything else?
